### DISCUSSION TITLE
Oauth Access Tokens revoked after only a few requests

### DISCUSSION THREAD

**Discussion by phillipdavidstearns** [etsy/open-api#1241] [Q&A] [answered]
After I successfully retrieve an access token, I am able to make a successful request to in-scope resources once or twice, maybe three or four more times, only to have that access token revoked before its expiry when making a similar request within the hour that token is issued.

I've also successfully obtained refreshed access tokens from the initial access token, but the same behavior follows.

Refreshing the token before each request doesn't resolve the issue either.

I'm working with Python3 Requests to make my server calls.

I'd like to know:

1. What would cause the OAuth access from being revoked, sometimes instants after being granted an access token?
2. Is there a known issue with using the Python3 Requests library to make calls to Etsy OpenAPI v3 endpoints? If so, are there guides addressing known issues?
3. Is there something I need to be doing to keep the connection alive other than refreshing the access tokens before the refresh token expiry?

**Comment by mjec** [comment]
Hi @phillipdavidstearns, this and the issue raised in [your other post](https://github.com/etsy/open-api/discussions/1240) are both caused by a bug on the Etsy side. We are working on a fix. Thank you for your helpful reports!

**Reply by phillipdavidstearns** [reply]
Thanks for your reply! I'm glad to know that it's not totally something I'm doing wrong. Please let me know where I can follow for updates. If there's any more detail that I could share with you to help bug hunt and squash, LMK!

**Reply by mjec** [reply, marked as answer]
Thanks for the offer! We've just rolled out a fix, and so you should be good to go. If you continue to experience issues please let us know.

**Reply by phillipdavidstearns** [reply]
Amazing! So far so good. Big thanks to you and the team for the quick fix.
